contracts/abstracts/Reservable.sol

```diff
@@ -70,21 +71,27 @@ contract Reservable is BeforeTransferERC20 {
     }
 
     function reserve(address sender, address recipient, address executor, uint256 amount, uint256 fee, uint256 nonce,
-        uint256 expiryBlockNum, bytes memory sig) public returns (bool success) {
+        uint256 expiryBlockNum, bytes memory sig) virtual public returns (bool success) {
+
         require(executor != address(0), "Reservable: cannot execute from zero address");
         require(expiryBlockNum > block.number, "Reservable: invalid block expiry number");
         require(_reserved[sender][nonce]._expiryBlockNum == 0, "Reservable: the sender used the nonce already");
 
-        uint256 total = amount.add(fee);
+        uint256 total;
+        unchecked {
+            total = amount + fee;
+        }
         require(total >= 0, "Reservable: invalid reserve amount");
         require(_unreservedBalance(sender) >= total, "Reservable: insufficient unreserved balance");
 
-        bytes32 hash = keccak256(abi.encodePacked(address(this), sender, recipient, executor, amount, fee, nonce, expiryBlockNum));
+        uint256 chainId = chainId();
+        bytes32 hash = keccak256(abi.encodePacked(GluwacoinModels.SigDomain.Reserve, chainId, address(this), sender, recipient, executor, amount, fee, nonce, expiryBlockNum));
         Validate.validateSignature(hash, sender, sig);
 
-        _reserved[sender][nonce] = Reservation(amount, fee, recipient, executor, expiryBlockNum,
-            ReservationStatus.Active);
-        _totalReserved[sender] = _totalReserved[sender].add(total);
+        _reserved[sender][nonce] = Reservation(amount, fee, recipient, executor, expiryBlockNum, ReservationStatus.Active);
+        unchecked {
```

Comment:
This one can't use unchecked, we must remove the unchecked so solidity will revert if there is overlow